internal/controller/oauth/base_provider.go

```diff
@@ -0,0 +1,50 @@
+package oauth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// BaseProvider implements common OAuth functionality
```

Comment:
It is not clear to me for the relationship between `BaseProvider` and `OidcProvider`. If it is common OAuth functions it is better to be defined as util functions.